daal4py/sklearn/neighbors/knn.py

```diff
@@ -163,6 +166,58 @@ def kneighbors(self, X=None, n_neighbors=None, return_distance=True):
 
 
 class KNeighborsClassifier(BaseKNeighborsClassifier, KNeighborsMixin):
+    def fit(self, X, y):
+        X, y = self._validate_data(X, y, accept_sparse="csr", multi_output=True)
+
+        try:
+            fptype = getFPType(X)
+        except ValueError:
+            fptype = None
+
+        numeric_type = True if np.issubdtype(X.dtype, np.number) and np.issubdtype(y.dtype, np.number) else False
+        single_output = False if y.ndim > 1 and y.shape[1] > 1 else True
+
+        if daal_check_version((2020, 3)) \
+        and self.weights in ['uniform', 'distance'] and self.algorithm in ['brute', 'kd_tree'] \
+        and (self.metric == 'minkowski' and self.p == 2 or self.metric == 'euclidean') \
+        and single_output and fptype is not None and not sp.issparse(X) and numeric_type:
+            logging.info("sklearn.neighbors.KNeighborsClassifier.fit: " + method_uses_daal)
+
+            y = make2d(y)
+            self.outputs_2d_ = False
+
+            self.classes_ = []
+            self._y = np.empty(y.shape, dtype=np.int)
+            for k in range(self._y.shape[1]):
+                classes, self._y[:, k] = np.unique(y[:, k], return_inverse=True)
+                self.classes_.append(classes)
+
+            self.classes_ = self.classes_[0]
+            self._y = self._y.ravel()
```

Comment:
Why store a training labels?

Reply:
Because it's using in stock `predict_proba()`